--- eshell.el.orig	2007-12-14 04:28:42.000000000 -0800
+++ eshell.el	2007-12-14 04:28:20.000000000 -0800
@@ -300,6 +300,9 @@
       (= (file-modes eshell-directory-name)
 	 eshell-private-directory-modes)))
 
+(defvar eshell-windowed nil
+  "*eshell will be displayed in another window, splitting if necessary")
+
 ;;;_* Running Eshell
 ;;
 ;; There are only three commands used to invoke Eshell.  The first two
@@ -332,7 +335,7 @@
     ;; it's necessary to add `eshell-buffer-name' to the variable
     ;; `same-window-buffer-names', which is done when Eshell is loaded
     (assert (and buf (buffer-live-p buf)))
-    (pop-to-buffer buf)
+    (pop-to-buffer buf eshell-windowed)
     (unless (eq major-mode 'eshell-mode)
       (eshell-mode))
     buf))
