? lisp/nxml/char-name/subdirs.el
Index: src/s/darwin.h
===================================================================
RCS file: /sources/emacs/emacs/src/s/darwin.h,v
retrieving revision 1.34
diff -u -r1.34 darwin.h
--- src/s/darwin.h	8 Jan 2008 20:45:11 -0000	1.34
+++ src/s/darwin.h	30 Apr 2008 17:33:44 -0000
@@ -276,7 +276,7 @@
    end of the header for adding load commands.  Needed for dumping.
    0x690 is the total size of 30 segment load commands (at 56
    each).  */
-#define LD_SWITCH_SYSTEM_TEMACS -prebind LIBS_CARBON -Xlinker -headerpad -Xlinker 690
+#define LD_SWITCH_SYSTEM_TEMACS -prebind LIBS_CARBON -Xlinker -headerpad -Xlinker 8c0
 
 #define C_SWITCH_SYSTEM_TEMACS -Dtemacs
 
